tests/test_cases/test_multi_trig_multi_coro/test_multi_trig_multi_coro.py

```diff
@@ -0,0 +1,58 @@
+import cocotb
+from cocotb.log import SimLog
+from cocotb.triggers import Timer, RisingEdge, FallingEdge
+from cocotb.result import TestFailure, TestSuccess
+
+import sys
+
+@cocotb.coroutine
+def clock_gen(signal, num):
+    for x in range(num):
+        signal <= 0
+        yield Timer(500)
+        signal <= 1
+        yield Timer(500)
+
+class DualMonitor:
+    def __init__(self, signal):
+        self.log = SimLog("cocotb.%s" % (signal._name))
+        self.monitor_edges = [0, 0]
+        self.signal = signal
+
+    @cocotb.coroutine
+    def signal_mon(self, signal, idx):
+        while True:
+            result = yield RisingEdge(signal)
+            self.log.info("signal_mon called on %s" % result)
+            self.monitor_edges[idx] += 1
+            
+    @cocotb.coroutine
+    def signal_mon_both(self, signal, idx):
+        while True:
```

Comment:
It would be good to clean up these coroutines. Can you add a `self.running` flag, which you clear at the end of `start` on line 46?